### ISSUE TITLE
Cannot run --tags=start after a recent commit

### ISSUE THREAD

**Issue by ewonchang** [spantaleev/matrix-docker-ansible-deploy#3161]
**Playbook Configuration**:
My `inventory/host_vars/matrix.domain.com/vars.yml` looks like this:
```yaml
matrix_domain: domain.com
matrix_coturn_turn_external_ip_address: x.x.x.x

matrix_homeserver_implementation: synapse
matrix_architecture: 'arm64'

matrix_synapse_federation_domain_whitelist:
- matrix.org

matrix_homeserver_generic_secret_key: 'xxxxx'

devture_postgres_connection_password: 'xxxxx'

matrix_static_files_container_labels_base_domain_enabled: true
matrix_static_files_file_index_html_enabled: false
matrix_static_files_container_labels_base_domain_root_path_redirection_enabled: false

matrix_synapse_ext_password_provider_shared_secret_auth_enabled: true
matrix_synapse_ext_password_provider_shared_secret_auth_shared_secret: xxxxx

matrix_playbook_reverse_proxy_type: playbook-managed-traefik
devture_traefik_config_certificatesResolvers_acme_email: acme_matrix@domain.com

matrix_mautrix_telegram_enabled: true
matrix_mautrix_telegram_api_id: xxxxx
matrix_mautrix_telegram_api_hash: xxxxx

matrix_heisenbridge_enabled: true
matrix_heisenbridge_identd_enabled: true

matrix_mautrix_whatsapp_enabled: true
```

**Host Configuration**
My `inventory/hosts` looks like this:
```yaml
[matrix_servers]
matrix.domain.com ansible_host=x.x.x.x ansible_connection=local ansible_ssh_user=ubuntu become=true become_user=root
```

**Matrix Server:**
 - OS: Ubuntu 22.04
 - Architecture: arm64

**Ansible:**
ansible-playbook [core 2.16.3] installed through `pip`

**Problem description**:

In the last few weeks, I cannot manually start (`--tags=start`) using the above configurations, which has been working fine for a few years. `--tags=setup-all` can run successfully, though. And if I reboot the server, Matrix will start automatically.

The output of `/usr/local/bin/ansible-playbook -i /home/ubuntu/matrix-docker-ansible-deploy/inventory/hosts /home/ubuntu/matrix-docker-ansible-deploy/setup.yml --tags=start`
```
...
TASK [custom/matrix-synapse : Check Matrix Client API] *****************************************************************************************************************************************
fatal: [matrix.domain.com -> 127.0.0.1]: FAILED! => changed=false
  elapsed: 0
  msg: 'Status code was -1 and not [200]: Request failed: <urlopen error [Errno 111] Connection refused>'
  redirected: false
  status: -1
  url: https://matrix.domain.com/_matrix/client/versions
...ignoring

TASK [custom/matrix-synapse : Fail if Matrix Client API not working] ***************************************************************************************************************************
fatal: [matrix.domain.com]: FAILED! => changed=false
  msg: 'Failed checking Matrix Client API is up at `matrix.domain.com` (checked endpoint: `https://matrix.domain.com/_matrix/client/versions`). Is Synapse running? Is port 443 open in your firewall? Full error: {''redirected'': False, ''url'': ''https://matrix.domain.com/_matrix/client/versions'', ''status'': -1, ''elapsed'': 0, ''changed'': False, ''failed'': True, ''msg'': ''Status code was -1 and not [200]: Request failed: <urlopen error [Errno 111] Connection refused>''}'

PLAY RECAP *************************************************************************************************************************************************************************************
matrix.domain.com       : ok=8    changed=0    unreachable=0    failed=1    skipped=4    rescued=0    ignored=1
```

Upon researching, I found this [issue](https://github.com/spantaleev/matrix-docker-ansible-deploy/issues/3095) similar to mine, and I tried to add the following line manually, to no avail (still the same error output)
```
matrix_synapse_container_additional_networks: ['traefik', 'matrix-postgres', 'matrix-exim-relay']
``` 

**Additional context**
- every component runs on a single server, a public VPS
- I have been fully migrated to traefik a few months ago, so the issue is less likely due to that. 
- I don't have any firewall preventing applications accessing 127.0.0.1 (at least I don't think)
- Also, there is a warning about Python crypt module in `--tags=setup-all`
```
TASK [galaxy/postgres : Ensure managed Postgres databases prepared] ****************************************************************************************************************************
[DEPRECATION WARNING]: Encryption using the Python crypt module is deprecated. The Python crypt module is deprecated and will be removed from Python 3.13. Install the passlib library for
continued encryption functionality. This feature will be removed in version 2.17. Deprecation warnings can be disabled by setting deprecation_warnings=False in ansible.cfg.
included: /home/ubuntu/matrix-docker-ansible-deploy/roles/galaxy/postgres/tasks/util/prepare_managed_databases.yml for matrix.domain.com
```

**Comment by spantaleev** [comment]
What happens after `--tags=start`? Which services are failing to start? Check their logs.

What you're showing us is the output of `--tags=self-check` which may be somewhat relevant, but it's not so interesting right now.

**Comment by ewonchang** [comment]
TLDR: problem solved; it's due to my operational error
---

You are absolutely right. I lumped `start` and `self-check` together without noticing it.
Running `start` alone give me `matrix-synapse.service` errors 
```
Feb 07 16:03:21 oc-matrix matrix-synapse[284187]: Error response from daemon: network matrix-homeserver not found
```
I tried to add `matrix_synapse_container_additional_networks: ['traefik', 'matrix-postgres', 'matrix-exim-relay','matrix-homeserver']` in my `vars.yml`, only to produce other networks not found error
I then checked docker network
```
$ sudo docker network ls
NETWORK ID     NAME              DRIVER    SCOPE
35daa12e9dd2   bridge            bridge    local
fa0fb898d331   host              host      local
582ee37f4690   matrix-postgres   bridge    local
80ab37510626   none              null      local
```
The networks of interest are not here! At this moment I realized that in my infitite wisdom, I did `sudo docker system prune` while Matrix is down, thus deleted "dangling" networks.

Solution:
- first, `stop-all`; 
- reset any failed systemd service `systemctl reset-failed`
- then `--tags=setup-all`, 
- finally `--tags=start`
- (optionally) `--tags=self-check`
